[dtensor] fix experimental_op slice_scatter

wanchaol opened the pull request:
Test Plan: test with spmd e2e flow

Differential Revision: D43740349

wanchaol commented:
#easycla

wanchaol commented:
/easycla

fegin reviewed (approved):
Thanks for the fix. Linter errors are real.

wanchaol commented:
/easycla

Commit:
[dtensor] fix experimental_op slice_scatter (#95894)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/95894

Test Plan: test with spmd e2e flow

Reviewed By: fegin

Differential Revision: D43740349

fbshipit-source-id: 4c2966d0b1ee5a64a41b86ca32fd73468f22413e
